src/libraries/Microsoft.Extensions.Logging.EventSource/src/EventSourceLogger.cs

```diff
@@ -57,23 +58,58 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             {
                 return;
             }
+
+            bool formattedMessageEventEnabled = _eventSource.IsEnabled(EventLevel.Critical, LoggingEventSource.Keywords.FormattedMessage);
+            bool messageEventEnabled = _eventSource.IsEnabled(EventLevel.Critical, LoggingEventSource.Keywords.Message);
+            bool jsonMessageEventEnabled = _eventSource.IsEnabled(EventLevel.Critical, LoggingEventSource.Keywords.JsonMessage);
+
+            if (!formattedMessageEventEnabled
+                && !messageEventEnabled
+                && !jsonMessageEventEnabled)
+            {
+                return;
+            }
+
             string? message = null;
 
+            Activity? activity = Activity.Current;
+            string? activityTraceId;
+            string? activitySpanId;
+            string? activityTraceFlags;
+            if (activity != null && activity.IdFormat == ActivityIdFormat.W3C)
```

Comment:
Are we sure we don't want support non W3C cases too?

Reply:
Well for non-W3C TraceId is going to spit out `00000000000000000000000000000000` and SpanId is going to spit out `0000000000000000`. We could just push `Activity.Id` into the events which would include `W3C`, `Hierarchical`, and (presumably) anything added in the future. BUT. Accessing `Activity.Id` will force an allocation in the process being monitored. And it will force consumers to have to inspect the value and understand the possible formats/differences. Probably also the consumers will have to chop up the `Id` into the individual components. In my world of OTel I don't need hierarchical but if you feel there is a need and the tradeoffs are OK, I'm happy to switch it.

Reply:
We have been supporting the `Hierarchical` cases with external logger scopes. 

https://github.com/dotnet/runtime/blob/b93061adc14eaaf5e83e366f9fba507ad917308c/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs#L291

Also, will it make sense to add the parent Id to the event as we do in scopes? 

CC @noahfalk if he has more insight if we should ignore the `Hierarchical` cases or not.

Reply:
Just FYI the reason I went with `ActivityTraceId`, `ActivitySpanId`, and `ActivityTraceFlags` is because those are the only things defined on the OTel Logs Data Model for correlation: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#log-and-event-record-definition

Happy to add other things, but from OTel perspective they aren't required.

Reply:
I'd be fine omitting support for hierarichal. Worst case I forsee is we get feedback that someone wants Hierarichal support, we decide the scenario in the request is important and then we add that too in a future release. I think that has a pretty low chance of happening.